Assets/Scripts/Interpreter/Lib/Wall.cs

```diff
@@ -0,0 +1,72 @@
+using Interpreter.Eval.Type;
+using UnityEngine;
+
+namespace Interpreter.Lib.Wall
+{
+    public class CreateWall : MonoBehaviour
+    {
+        public GameObject obj;
+        public Nil Exec(Number x1, Number y1, Number x2 = null, Number y2 = null)
+        {
+            //@TODO: check for resources
+            
+            float x1f = (float) x1.AsNumber();
+            float y1f = (float) y1.AsNumber();
+            
+            if (x2 == null || y2 == null)
```

Comment:
Essa condição não é necessária porque o interpretador não deixará o usuário rodar o código se o argumento não for passado. Ou seja, `x2` e `y2` nunca serão nulos.

Reply:
Se for necessário passar argumentos nulos, eu sugiro criar uma outra função, não terei tempo para acrescentar suporte a parâmetros opcionais no interpretador.

Reply:
Show, vou consertar e atualizar aq